### PR TITLE
feat: local configuration

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request adds initial support for localized configuration files using `.example.tsx` syntax.

Resolves #91

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -1,4 +1,23 @@
...
+	return map;
+}, {});
+
+const config: Config = {
+	...baseConfig,
+	widgets: {
+		...baseConfig.widgets,
+		...widgetConfig
+	}
+};
```
Comment (bitpshr) [inline src/main.tsx:21]: Note that mixing both global widget configuration and local `.example.ts` configuration is possible for the sake of as much compatibility as possible, with localized configurations taking precedence.

**Review by matt-gadd — APPROVED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -40,18 +43,15 @@ function getReadmeFileNames(config: any): string[] {
 	return filenames;
 }
 
-function getExampleFileNames(config: any): string[] {
+function getExampleFileNames(config: Config): string[] {
 	const filenames: string[] = [];
+	if (!config.widgets) return filenames;
```
Comment (agubler) [inline src/App.tsx:48]: We tend to always use curly braces `{` with if

File: src/App.tsx [hunk 3]
```diff
@@ -40,18 +43,15 @@ function getReadmeFileNames(config: any): string[] {
...
-		const widget = config.widgets[key];
-		if (widget.overview && widget.overview.example) {
-			filenames.push(config.examplePath(key, widget.overview.example.filename));
-		}
-		if (widget.examples) {
-			widget.examples.forEach((example: any) => {
-				filenames.push(config.examplePath(key, example.filename));
-			});
-		}
+		if (!config.widgets) return;
```
Comment (agubler) [inline src/App.tsx:50]: would this ever be trie with the if above outside the forEach?

File: src/Example.tsx [hunk 4]
```diff
@@ -58,11 +59,12 @@ export default factory(function Example({
...
-		? config.widgets[widgetName].overview.example
-		: config.widgets[widgetName].examples.find(
-				(e: any) => e.filename.toLowerCase() === exampleName
-		  );
+	const example =
+		config.widgets &&
+		config.widgets[widgetName].find((e: any) =>
+			isOverview ? e.overview : e.filename.toLowerCase() === exampleName
+		);
+	if (!example) return;
```
Comment (agubler) [inline src/Example.tsx:67]: Same comment re braces 
```ts
if (!example) {
    return;
}
```

File: src/ExampleSandbox.tsx [hunk 5]
```diff
@@ -26,16 +26,12 @@ export default factory(function Example({ properties, middleware: { theme, resiz
...
-	const example = isOverview
-		? config.widgets[widgetName].overview.example
-		: config.widgets[widgetName].examples.find(
-				(e: any) => e.filename.toLowerCase() === exampleName
-		  );
-	return (
+	const example =
+		config.widgets &&
+		config.widgets[widgetName].find((e: any) => e.filename.toLowerCase() === exampleName);
+	return example ? (
```
Comment (agubler) [inline src/ExampleSandbox.tsx:32]: could this be `example && (...)`?

File: src/App.tsx (hunk 3 above)
Comment (bitpshr) [reply src/App.tsx:50]: No, this was only to satisfy typings.

File: src/ExampleSandbox.tsx (hunk 5 above)
Comment (bitpshr) [reply src/ExampleSandbox.tsx:32]: I like this much more as well, but this does not satisfy typings in this case. I didn't dig much to see why though.

**Pull request merged**